### DISCUSSION TITLE
DAS keeps dropping-out...

### DISCUSSION THREAD

**Discussion by FA-Bubba** [SamDel/ChromeCast-Desktop-Audio-Streamer#158] [General]
I was still using DAS version 5.1.0, and today, it kept dropping out, so I upgraded to version 5.4.0, but it still constantly drops-out, and at times ceases to respond.  I have attached some screen shots,and the Log.  Please advise what I can do to get it to be more stable.

[DAS-Log-2024-06-01.TXT](https://github.com/user-attachments/files/15523499/DAS-Log-2024-06-01.TXT)
![ScreenShot-v5-4-0--2024-06-01-b](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/assets/4895674/6a36dfd9-ecfe-41db-b445-b561a0266d4e)
![DAS-NotResponding-2024-06-01](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/assets/4895674/414a9804-2291-413b-a0bb-8470b64d7f4b)
![ScreenShot-v5-4-0--2024-06-01](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/assets/4895674/30bb45f8-9e8d-49b3-a2d3-ffd85d9080e0)

**Comment by SamDel** [comment]
I see some timeouts in the log. How's the network load?

For the not responding application: Can you have a look if there's a line in the Windows Event Viewer (Windows Logs - Application)?

**Comment by FA-Bubba** [comment]
The network load has been consistent for the past 5 years....

As for the Windows Logs, I'm not sure where you want me to look, but I did see the following listed six times in the past two days:

The program ChromeCast.Desktop.AudioStreamer.exe version 5.4.0.0 stopped interacting with Windows and was closed. To see if more information about the problem is available, check the problem history in the Security and Maintenance control panel.
 Process ID: 38b4
 Start Time: 01dab46de6139b7e
 Termination Time: 1342
 Application Path: C:\Program Files (x86)\Desktop Audio Streamer\Desktop Audio Streamer\ChromeCast.Desktop.AudioStreamer.exe
 Report Id: 9ea3df5e-2a3c-4e07-8f39-ecfdd7fe934f
 Faulting package full name: 
 Faulting package-relative application ID: 
 Hang type: Unknown

Please let me know if there is a specific section of the Windows Logs where you want me to look.

Thank you!

**Comment by SamDel** [comment]
A good test is to change to mp3 streaming (temporary). Then we know it's load related, or something else.

I don't think there are more details in the Windows Logs. When the application crashes (not hangs) it writes more details.

**Comment by FA-Bubba** [comment]
Thanks for the follow-up....
After the 6th Windows failure, DAS has been running as expected for the past 36 hours, so I'll leave it as is for now.

**Comment by FA-Bubba** [comment]
PS:  ALL of the music I stream are MP3s via iTunes...

**Comment by SamDel** [comment]
For the performance of your home network it's more important that the format that's used for streaming between DAS and the devices is Wav 48000 16 bit. The log says:

>Stream format set to Pcm 48000 16 bit

For better performance on your home network you can change that to mp3 128/320 kbps on the options tab:

![afbeelding](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/assets/25846417/699a8e0e-ae72-4fd2-af1c-da8e1209c99f)

**Reply by FA-Bubba** [reply]
I changed it to MP3 (320 kbps) a LONG time ago

**Reply by SamDel** [reply]
You're right, I was looking at the wrong line in the log.

> Set stream format to Mp3_320

**Comment by FA-Bubba** [comment]
DAS continues to drop-out- and frequently goes into Not Responding mode.  As recently as April 2024, it streamed flawlessly 24x7, but not dies after a few hours.  I just restarted it, and am trying to Cast to just 3 speakers, but one set fails to Stream.  See screenshot:
![ScreenShot-2024-06-30](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/assets/4895674/b7c22263-d083-4ed3-a60e-70d90f118fdb)

also, here's the log:
[DAS-Log-2024-06-30.txt](https://github.com/user-attachments/files/16041972/DAS-Log-2024-06-30.txt)


...and a previous log from a few weeks ago:
[DAS-Log-2024-06-01.TXT](https://github.com/user-attachments/files/16041973/DAS-Log-2024-06-01.TXT)
![DAS-NotResponding-2024-06-01](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/assets/4895674/039f79df-19f3-438b-9515-f1afa966bcf2)

![ScreenShot-v5-4-0--2024-06-01](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/assets/4895674/5641057d-d5c3-4764-9fe2-e392acb8e312)

How can this be corrected?

**Comment by FA-Bubba** [comment]
Here's a new Log, just after the DAS began to fail again...

[DAS-Log-2024-06-30--2.txt](https://github.com/user-attachments/files/16042071/DAS-Log-2024-06-30--2.txt)

**Comment by SamDel** [comment]
There are a lot of exceptions in the logs related to the connection to the devices (to send control messages). [Back in 2021](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/issues/84#issuecomment-945717231) there were none.

It can be that one of the fixes that were made since cause your drop-outs. I will look into that later. 

Maybe you can try an older version meanwhile (like [version 3.9](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/releases/tag/v3.9)). I wonder if that version works better for you!?

**Comment by FA-Bubba** [comment]
I installed v 3.9, but when I launch it v5.4.0.0 is displayed on the Options page.  Do I need to uninstall first?  If so, what are the best steps so I don't lose my Devices?

**Comment by SamDel** [comment]
Yes, you have to uninstall first when downgrading. 
The settings from the last time you used 3.9 are loaded. If the 3.9 settings are not there the default settings are loaded. Devices are automatically discovered (and are also in the settings file when something goes wrong with discovering).

When you upgrade to 5.4 after the test your current settings are loaded again.

**Comment by FA-Bubba** [comment]
OK... I tried to uninstall 3.9 first, but that did not seem to work, so I reinstalled 5.0.4,  Then I uninstalled that and installed 3.9, but it will not start.   I tried several times.

What would you suggest?

**Comment by SamDel** [comment]
It's easier to install [5.4.1.zip](https://github.com/user-attachments/files/16046268/5.4.1.zip) then. 
It's the same as 5.4, I only reverted most patches to the [device connection](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/blob/b94e235e8420cc3066025d64c43ee4a59f194ca7/Source/ChromeCast.Desktop.AudioStreamer/Communication/DeviceConnection.cs) since 3.9.

**Comment by FA-Bubba** [comment]
Thank you!  I am back to ver 3.9....  I had to reinstall ver 5.4.2, the uninstall it and then install 3.9.  Here's a screen shot:
![ScreenShot-2024-06-30-ver3-9](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/assets/4895674/89686ac3-8e60-42c4-ad2c-9840d94dfab2)

The previously unstable devices are working fine now; hopefully, they will continue 24x7.

Thanks for your help.

**Comment by FA-Bubba** [comment]
Since switching to version 3.9 five days ago, DAS has been streaming to ALL of my speakers 24x7 with no dropouts.

**Comment by SamDel** [comment]
Nice! Can you also try version 5.4.1 I posted before, to pinpoint if it's due to the changes to the device connection?

**Comment by FA-Bubba** [comment]
Version 5.4.1 is not on the Releases page.

**Comment by SamDel** [comment]
This one: [5.4.1.zip](https://github.com/user-attachments/files/16046268/5.4.1.zip).
It's not an official release, it's a version to test with. It's the same as 5.4, I only reverted most patches to the [device connection](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/blob/b94e235e8420cc3066025d64c43ee4a59f194ca7/Source/ChromeCast.Desktop.AudioStreamer/Communication/DeviceConnection.cs) since 3.9.

**Comment by FA-Bubba** [comment]
Version 5.4.1 drops out frequently.  Log is attached.
[DAS-Log-2024-07-07.txt](https://github.com/user-attachments/files/16119676/DAS-Log-2024-07-07.txt)

**Comment by SamDel** [comment]
A lot of drop-outs. But no errors on the device connection in this log. 
Have you tried running 5.4.1 for a longer period?

**Comment by FA-Bubba** [comment]
No, I cannot deal with the drop-outs...  v 3.9 works great for me.
